joomla/joomla-cms
# Disable sql query cache in debug mode

Pull Request for Issue https://github.com/joomla/joomla-cms/issues/13265.

### Summary of Changes

This PR makes the query cache disabled when db driver runs in debug mode.
Only for mysql, because the profiling only seems to work there. 

Also add checks for profiling in pdomysql (was missing), which meand pdo mysql driver now as the "Profile" in debug console too.

### Testing Instructions

1. Apply patch
2. Enable debug mode in global config and system debug plugin
3. Test Mysql, MySqli and PDO Mysql drivers: Reload the pages and confirm in debug console "Database Queries" "Profile" that each query in run without using db cache

#### Example

With query cache
![image](https://cloud.githubusercontent.com/assets/9630530/21296453/ee2ca178-c563-11e6-8f26-7370e7d1872b.png)

Without query cache
![image](https://cloud.githubusercontent.com/assets/9630530/21296451/e42634aa-c563-11e6-9bee-17c7bfcedf91.png)

### Documentation Changes Required

None.

### Notes

- Didn't test "mysql" driver because is not available in my test server (only "mysqli" and "pdo (mysql)")
- before merging, this should be reviewed from a expert db person.
- More info http://dev.mysql.com/doc/refman/5.7/en/query-cache-configuration.html
